examples/dcn.py

```diff
@@ -0,0 +1,571 @@
+combination strategies exist: a stacked structure, where the deep network is
+placed on top of the cross network, and a parallel structure, where they
+operate in parallel.
+
+![Parallel Layers](http://drive.google.com/uc?export=view&id=1WtDUCV6b-eetUnWVCAmcPh8mJFut5EUd)
+![Stacked Layers](http://drive.google.com/uc?export=view&id=1xo_twKb847hasfss7JxF0UtFX_rEb4nt)
+
+Now that we know a little bit about DCN, let's start writing some code. We will
+first train a DCN on a toy dataset, and demonstrate that the model has indeed
+learnt important feature crosses.
+
+Let's get our imports sorted.
+"""
+
+import keras
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+import keras_rs
+
+"""
+Let's also define variables which will be reused throughout the example.
+"""
+
+TOY_CONFIG = {
+    "learning_rate": 0.01,
+    "num_epochs": 100,
+    "batch_size": 1024,
+}
+
+MOVIELENS_CONFIG = {
+    # features
+    "int_features": [
+        "movie_id",
+        "user_id",
+        "user_gender",
+        "bucketized_user_age",
+    ],
+    "str_features": [
+        "user_zip_code",
+        "user_occupation_text",
+    ],
+    # model
+    "embedding_dim": 32,
+    "deep_net_num_units": [192, 192, 192],
+    "projection_dim": 20,
+    "dcn_num_units": [192, 192],
+    # training
+    "learning_rate": 0.01,
+    "num_epochs": 10,
+    "batch_size": 1024,
+}
+
+LOOKUP_LAYERS = {
+    "int": keras.layers.IntegerLookup,
+    "str": keras.layers.StringLookup,
+}
+
+"""
+Here, we define a helper function for visualising weights of the cross layer in
+order to better understand its functioning. Also, we define a function for
+compiling, training and evaluating a given model.
+"""
+
+
+def visualize_layer(matrix, features):
+    plt.figure(figsize=(9, 9))
+
+    im = plt.matshow(np.abs(matrix), cmap=plt.cm.Blues)
+
+    ax = plt.gca()
+    divider = make_axes_locatable(plt.gca())
+    cax = divider.append_axes("right", size="5%", pad=0.05)
+    plt.colorbar(im, cax=cax)
+    cax.tick_params(labelsize=10)
+    ax.set_xticklabels([""] + features, rotation=45, fontsize=10)
+    ax.set_yticklabels([""] + features, fontsize=10)
+
+
+def train_and_evaluate(
+    learning_rate,
+    epochs,
+    train_data,
+    test_data,
+    model,
+):
+    optimizer = keras.optimizers.AdamW(learning_rate=learning_rate)
+    loss = keras.losses.MeanSquaredError()
+    rmse = keras.metrics.RootMeanSquaredError()
+
+    model.compile(
+        optimizer=optimizer,
+        loss=loss,
+        metrics=[rmse],
+    )
+
+    model.fit(
+        train_data,
+        epochs=epochs,
+        verbose=0,
+    )
+
+    results = model.evaluate(test_data, return_dict=True, verbose=0)
+    rmse_value = results["root_mean_squared_error"]
+
+    return rmse_value, model.count_params()
+
+
+def print_stats(rmse_list, num_params, model_name, num_trials=1):
```

Comment:
Can't you just look at the length of `rmse_list` instead of having this extra `num_trials` argument.

The main reason is that 10 is hardcoded in 2 places and it's not intuitive why you're passing 10 to `print_stats` further down.